privacy_partner_report/wizard/privacy_report_partner.py

```diff
@@ -0,0 +1,246 @@
+                return True
+        return False
+
+    def _get_default_table(self, name, data):
+        if data:
+            data_type = data[0][4]
+            res = self.env[data[0][1]]
+            for t in data:
+                res |= self.env[t[1]].browse(t[3])
+            if res:
+                values = {
+                    'name': name,
+                    'model_id': self.env['ir.model'].search(
+                        [('model', '=', res._name)]).id,
+                    'count_rows': len(res.ids),
+                    'type': data_type,
+                }
+                return values
+        return {}
+
+    def _get_model_from_table(self, table, partner):
+        new_tables = {}
+        for model in table.model_id:
+            rows = self._get_rows_from_model(model, partner)
+            new_tables[str(model.display_name)] = rows
+        return new_tables
+
+    def _get_rows_from_model(self, model, partner):
+        cr = self.env.cr
+        lines = self.env[model.model]
+        columns = [k for k, v in self.env[model.model]._fields.items()
+                   if v.comodel_name == 'res.partner' and
+                   not self._exclude_column(model.model, k)]
+        for column in columns:
+            lines |= self.env[model.model].search([(column, '=', partner.id)])
+        line_ids = ', '.join([str(i) for i in lines.ids])
+        query = "SELECT * FROM %s WHERE id IN (%s)" % (
+            model.model.replace('.', '_'), line_ids)
+        cr.execute(query)
+        rows = cr.dictfetchall()
+        rows = self._clean_data(model.model, rows)
+        return rows
+
+    def _get_rows_from_tables(self, tables, partner):
+        new_tables = {}
+        for table in tables:
+            data_table = self._get_model_from_table(table, partner)
+            new_tables[str(table.name)] = data_table
+        return new_tables
+
+    def _get_table_names(self, data):
+        names = []
+        for t in data:
+            if t[3] and not t[5] and t[0] not in names:
+                names.append(t[0])
+        return names
+
+    def _get_tables_from_partner(self, partner):
+        tables = [t[0] for t in [
+            [[self.env[m]._table, m, k, self.env[m].sudo().search(
+                [(k, '=', partner.id)]).ids, v.type, self.env[m]._transient]
+             for k, v in self.env[m]._fields.items()
+             if v.comodel_name == 'res.partner' and self.env[m]._auto and
+             not self._exclude_column(m, k)]
+            for m in [x for x in self.env.registry.keys()]] if t]
+        for i, t in enumerate(tables):
+            if t[4] == 'many2many':
+                if t[3]:
+                    relation = self.env[t[1]]._fields[t[2]].relation
+                    if relation:
+                        tables[i][0] = relation
+        return tables
+
+    def _print_report(self, data, xlsx_report=False):
+        records = self.env[data['model']].browse(data.get('ids', []))
```

Comment:
you should do the browse using `sudo()`. Otherwise you'll face zillons of errors due to missing permissions.